platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -294,6 +300,10 @@ export default class HangingProtocolService extends PubSubService {
     }
 
     this.protocols.set(protocolId, protocol);
+
+    // also insert it in the copy of the protocols map object, but make a deep
+    // copy of the protocol object first
+    this.protocolsCopy.set(protocolId, this._copyProtocol(protocol));
```

Comment:
This should be done on use/starting to update the given protocol, so that a copy is made as required, rather than copying everything.  Since the protocols can be reasonably large, it reduces startup cost by not copying until needed - and indeed, not until usable.